vercel/data_source_project_rolling_release.go

```diff
@@ -0,0 +1,234 @@
+	resp.TypeName = req.ProviderTypeName + "_project_rolling_release"
+}
+
+func (d *projectRollingReleaseDataSource) Configure(ctx context.Context, req datasource.ConfigureRequest, resp *datasource.ConfigureResponse) {
+	// Prevent panic if the provider has not been configured.
+	if req.ProviderData == nil {
+		return
+	}
+
+	client, ok := req.ProviderData.(*client.Client)
+	if !ok {
+		resp.Diagnostics.AddError(
+			"Unexpected Resource Configure Type",
+			fmt.Sprintf("Expected *client.Client, got: %T. Please report this issue to the provider developers.", req.ProviderData),
+		)
+		return
+	}
+
+	d.client = client
+}
+
+func (d *projectRollingReleaseDataSource) Schema(ctx context.Context, _ datasource.SchemaRequest, resp *datasource.SchemaResponse) {
+	resp.Schema = schema.Schema{
+		MarkdownDescription: "Data source for a Vercel project rolling release configuration.",
+		Attributes: map[string]schema.Attribute{
+			"project_id": schema.StringAttribute{
+				MarkdownDescription: "The ID of the project.",
+				Required:            true,
+			},
+			"team_id": schema.StringAttribute{
+				Optional:    true,
+				Computed:    true,
+				Description: "The ID of the Vercel team.",
+			},
+			"advancement_type": schema.StringAttribute{
+				MarkdownDescription: "The type of advancement for the rolling release. Either 'automatic' or 'manual-approval'.",
+				Computed:            true,
+			},
+			"stages": schema.ListNestedAttribute{
+				MarkdownDescription: "The stages for the rolling release configuration.",
+				Computed:            true,
+				NestedObject: schema.NestedAttributeObject{
+					Attributes: map[string]schema.Attribute{
+						"target_percentage": schema.Int64Attribute{
+							MarkdownDescription: "The percentage of traffic to route to this stage.",
+							Computed:            true,
+						},
+						"duration": schema.Int64Attribute{
+							MarkdownDescription: "The duration in minutes to wait before advancing to the next stage. Present for automatic advancement type.",
+							Computed:            true,
+						},
+					},
+				},
+			},
+		},
+	}
+}
+
+// ProjectRollingReleaseDataSourceModel reflects the structure of the data source.
+type ProjectRollingReleaseDataSourceModel struct {
+	AdvancementType types.String `tfsdk:"advancement_type"`
+	Stages          types.List   `tfsdk:"stages"`
+	ProjectID       types.String `tfsdk:"project_id"`
+	TeamID          types.String `tfsdk:"team_id"`
+}
+
+func (d *projectRollingReleaseDataSource) Read(ctx context.Context, req datasource.ReadRequest, resp *datasource.ReadResponse) {
+	var data ProjectRollingReleaseDataSourceModel
+
+	resp.Diagnostics.Append(req.Config.Get(ctx, &data)...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	var out client.RollingReleaseInfo
+	var err error
+	var convertedData ProjectRollingReleaseDataSourceModel
+	var diags diag.Diagnostics
+
+	// Retry up to 10 times (10s) for eventual consistency
```

Comment:
I don't follow? Why do we need this? 

If we do need, can we use the `vercel/retry.go` helper instead?